### PR TITLE
RDM-7631: Migrate Definition Store to OIDC

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-7631

### Change description ###

OpenID changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
     protected static final String ROLE_CCD_IMPORT = "ccd-import";
     private static final String REGEX_URI_IMPORT = "^" + ImportController.URI_IMPORT  + "/?$";
 
-    @Value("#{'${casedefinitionstore.authorised.services}'.split(',')}")
+    @Value("#{'${idam.s2s-authorised.services}'.split(',')}")
```
Comment (smathangi) [inline application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java]: remove whole bean .. now s2s validation happening in the serviceAuthFilter

File: application/src/main/resources/application.properties [hunk 2]
```diff
@@ -26,14 +26,28 @@ spring.jpa.properties.hibernate.ejb.interceptor=uk.gov.hmcts.ccd.definition.stor
...
+
 idam.s2s-auth.url=${IDAM_S2S_URL:http://localhost:4502}
 idam.s2s-auth.microservice=ccd_definition
 idam.s2s-auth.totp_secret=${DEFINITION_STORE_IDAM_KEY:AAAAAAAAAAAAAAAA}
 
-casedefinitionstore.authorised.services=${DEFINITION_STORE_S2S_AUTHORISED_SERVICES:ccd_data,ccd_gw,ccd_admin,jui_webapp}
+idam.s2s-authorised.services=${DEFINITION_STORE_S2S_AUTHORISED_SERVICES:ccd_data,ccd_gw,ccd_admin,jui_webapp}
+
+#max size of cache map in terms of number of entries
+definition.cache.max.size=${DEFINITION_CACHE_MAX_SIZE:5000}
```
Comment (smathangi) [inline application/src/main/resources/application.properties]: remove these definition.cache props.. they are data-store ones

File: build.gradle [hunk 3]
```diff
@@ -208,11 +209,20 @@ subprojects { subproject ->
...
+        compile group: 'uk.gov.hmcts.reform', name: 'idam-client', version: '1.3.1'
+
+        compile "org.springframework.security:spring-security-web"
+        compile "org.springframework.security:spring-security-config"
+        compile "org.springframework.boot:spring-boot-starter-oauth2-client"
+        compile "org.springframework.boot:spring-boot-starter-oauth2-resource-server"
+        compile group: 'com.nimbusds', name: 'nimbus-jose-jwt', version: '7.9'
+        implementation "io.github.openfeign:feign-httpclient:10.7.0"
+
+        compile group: 'com.hazelcast', name: 'hazelcast'
```
Comment (smathangi) [inline build.gradle]: remove hazelcast

File: charts/ccd-definition-store-api/values.yaml [hunk 4]
```diff
@@ -21,6 +21,9 @@ java:
     DEFINITION_STORE_DB_OPTIONS: "?sslmode=require"
     DEFINITION_STORE_DB_MAX_POOL_SIZE: 32
 
+    DEFINITION_CACHE_MAX_SIZE: 5000
```
Comment (smathangi) [inline charts/ccd-definition-store-api/values.yaml]: remove these..

File: infrastructure/main.tf [hunk 5]
```diff
@@ -115,9 +117,12 @@ module "case-definition-store-api" {
     DEFINITION_STORE_DB_PASSWORD = "${module.definition-store-db.postgresql_password}"
     DEFINITION_STORE_DB_OPTIONS = "?sslmode=require"
     DEFINITION_STORE_DB_MAX_POOL_SIZE = "${var.database_max_pool_size}"
+    DEFINITION_CACHE_MAX_SIZE           = "${var.definition_cache_max_size}"
```
Comment (smathangi) [inline infrastructure/main.tf]: remove these `DEFINITION_CACHE_MAX_SIZE`

File: infrastructure/variables.tf [hunk 6]
```diff
@@ -65,6 +65,16 @@ variable "appinsights_instrumentation_key" {
   default = ""
 }
 
+variable "definition_cache_max_size" {
```
Comment (smathangi) [inline infrastructure/variables.tf]: remove these pls

File: integration-tests/src/test/java/uk/gov/hmcts/integration/IntegrationTest.java [hunk 7]
```diff
@@ -50,39 +35,39 @@
...
-        private DataSource dataSource(final EmbeddedPostgres pg) throws SQLException {
-            final Properties props = new Properties();
-            // Instruct JDBC to accept JSON string for JSONB
-            props.setProperty("stringtype", "unspecified");
-            final Connection connection = DriverManager.getConnection(pg.getJdbcUrl("postgres", "postgres"), props);
-            LOG.info("Started Postgres, port number = {}", pg.getPort());
-            return new SingleConnectionDataSource(connection, true);
-        }
-    }
+//    @TestConfiguration
```
Comment (smathangi) [inline integration-tests/src/test/java/uk/gov/hmcts/integration/IntegrationTest.java]: any reason for commented out..?

File: application/src/main/resources/application.properties (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply application/src/main/resources/application.properties]: removed

File: build.gradle (hunk 3 above)
Comment (kiran-yenigala-hmcts) [reply build.gradle]: removed

File: charts/ccd-definition-store-api/values.yaml (hunk 4 above)
Comment (kiran-yenigala-hmcts) [reply charts/ccd-definition-store-api/values.yaml]: removed

File: infrastructure/main.tf (hunk 5 above)
Comment (kiran-yenigala-hmcts) [reply infrastructure/main.tf]: removed

File: infrastructure/variables.tf (hunk 6 above)
Comment (kiran-yenigala-hmcts) [reply infrastructure/variables.tf]: removed

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java]: what do you mean by whole bean.. do you mean AuthCheckerConfiguration

File: build.gradle (hunk 3 above)
Comment (smathangi) [reply build.gradle]: Still exists

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java (hunk 1 above)
Comment (smathangi) [reply application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java]: `authorizedServicesExtractor` bean

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/CacheConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+@EnableCaching
+@Configuration
+public class CacheConfiguration {
+
+    @Value("${user.cache.ttl.secs}")
+    private Integer userCacheTTLSecs;
+
+    @Bean
+    public CacheManager cacheManager() {
+        CaffeineCacheManager cacheManager = new CaffeineCacheManager("userInfoCache");
```
Comment (smathangi) [inline application/src/main/java/uk/gov/hmcts/ccd/definition/store/CacheConfiguration.java]: I guess spring boot has already this class with customisation via properties.. 
`CaffeineCacheManager`
https://docs.spring.io/spring-boot/docs/2.1.6.RELEASE/reference/html/boot-features-caching.html#boot-features-caching-provider-caffeine

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java [hunk 9]
```diff
@@ -1,57 +0,0 @@
...
-    @Bean
-    public Function<HttpServletRequest, Optional<String>> userIdExtractor() {
-        return request -> Optional.empty();
-    }
-
-    private Collection<String> extractRoleFromRequest(final HttpServletRequest request) {
-        if (request.getRequestURI().matches(REGEX_URI_IMPORT)) {
-            return Collections.singletonList(ROLE_CCD_IMPORT);
-        }
-        if (adminWebAuthorizationProperties.isEnabled()
```
Comment (smathangi) [inline application/src/main/java/uk/gov/hmcts/ccd/definition/store/AuthCheckerConfiguration.java]: Currently it's disabled in prod/aat/demo.. so unused/ uncompleted functionality. So, removed this whole class. The other check `ROLE_CCD_IMPORT` moved to securityConfiguration class

**Pull request merged**